app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -19,14 +19,33 @@ def send_code
       if result.success?
         handle_valid_otp_delivery_preference(user_selected_otp_delivery_preference)
       else
-        redirect_to user_two_factor_authentication_url(reauthn: reauthn?)
+        handle_invalid_otp_delivery_preference(result)
       end
     rescue Twilio::REST::RestError => exception
       invalid_phone_number(exception)
     end
 
     private
 
+    def validate_otp_delivery_preference_and_send_code
```

Comment:
Have you tried pulling this into `#send_code` as well? This looks like it's duplicating a lot of the login that is going on there.

Reply:
Yeah, I thought about it but wanted to keep it simple for now, and Code Climate doesn't seem to mind. What do you think? Can we refactor later?

Reply:
Up to you, just wanted to point it out to make sure you'd thought about it.